### PR TITLE
JS API/Core/Hooks: New - Get hook by id, after requirement of disable.

### PR DESCRIPTION
**Pull request by iNewLegend**
Usage:
```
$e.hooks.data = for data hooks.
$e.hooks.ui = for ui hooks.

$e.hooks.data.get('create-section-columns').deactivate();
$e.hooks.data.get('create-section-columns').activate();
```

### CODE REVIEW

File: core/common/assets/js/api/core/hooks/base.js [hunk 1]
```diff
@@ -69,6 +69,10 @@ export default class HooksBase extends elementorModules.Module {
 		elementorModules.forceMethodImplementation();
 	}
 
+	get( id ) {
```
Comment (kobizz) [inline core/common/assets/js/api/core/hooks/base.js:72]: You can use an object to store the `callbacksFlatList` with callback id as a key, then you don't need to walk through all callbacks to find the desired one

**Review by kobizz — CHANGES_REQUESTED** [review]

File: core/common/assets/js/api/core/hooks/base.js (hunk 1 above)
Comment (iNewLegend) [reply core/common/assets/js/api/core/hooks/base.js:72]: Handled: https://github.com/elementor/elementor/pull/12097/commits/c8e075954e693018ff59a1d58e6e5d4153ea3319

**Review by matipojo — APPROVED** [review]

File: core/common/assets/js/api/core/hooks/base.js [hunk 2]
```diff
@@ -278,7 +290,7 @@ export default class HooksBase extends elementorModules.Module {
 			this.callbacks[ event ][ command ].all.push( callback );
 		}
 
-		this.callbacksFlatList.push( callback );
+		this.callbacksFlatList[ callback.id ] = callback;
```
Comment (matipojo) [inline core/common/assets/js/api/core/hooks/base.js:293]: Use it as an object in all places.

**Review by matipojo — CHANGES_REQUESTED** [review]

File: core/common/assets/js/api/core/hooks/base.js (hunk 2 above)
Comment (iNewLegend) [reply core/common/assets/js/api/core/hooks/base.js:293]: Handled.

**Review by matipojo — DISMISSED** [review]

**Review by matipojo — CHANGES_REQUESTED** [review]

File: core/common/assets/js/api/core/hooks/base.js [hunk 3]
```diff
@@ -69,18 +69,22 @@ export default class HooksBase extends elementorModules.Module {
...
-	 * @param {boolean} flat
+	 * @param {boolean} flat - Return flat array.
 	 *
-	 * @returns {{}}
+	 * @returns {({}|array)}
 	 */
 	getAll( flat = false ) {
 		if ( flat ) {
-			return this.callbacksFlatList;
+			return Object.values( this.callbacksFlatList );
```
[the remark this answers is not in the document]
Comment (iNewLegend) [reply core/common/assets/js/api/core/hooks/base.js]: Handled

[the remark this answers is not in the document]
Comment (iNewLegend) [reply core/common/assets/js/api/core/hooks/base.js]: Handled

**Review by matipojo — APPROVED** [review]

**Review by kobizz — APPROVED** [review]

**Pull request merged**